.github/workflows/prover.yml

```
@@ -0,0 +1,47 @@
+name: Run prover guest code checks
+
+on:
+  pull_request:
+  merge_group:
+  push:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Rust
+        uses: dtolnay/rust-toolchain@nightly
+        with:
+          toolchain: nightly-2024-07-27
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
```

Final nit
In every step in this file you're doing with `name`.
Why in this it is straight into `uses`?
And why this is the only step that is not surrounded by empty lines?

---

>In every step in this file you're doing with name.
I've seen the same pattern in `functional.yml`

>And why this is the only step that is not surrounded by empty lines?
I treat it like a code. Usually, block of the code that delivers some logic goes line-by-line and blocks of logic are separated by an empty line. With that said, I think setting up rust and configuring its cache is the same "logical block", while, installing `clang` / `mold` / `sp1` are separate "logical steps/blocks"

Also I've seen the same in `functional.yml`.

Feel free to counter it and tell me if it's ugly, I'd be happy to fix it.

---

```suggestion

      - name: Rust cache
        uses: Swatinem/rust-cache@v2
        with:
          cache-on-failure: true
```
Let's just be consistent in this file. We can standardize later if necessary across the YAML CI files.